source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu

```diff
@@ -352,6 +352,134 @@ __global__ void cal_vq_deri(
 }
 
 
+
+template <typename FPTYPE>
+__global__ void cal_stress_drhoc_aux0(
+        const FPTYPE* r, const FPTYPE* rhoc, 
+        const FPTYPE *gx_arr, const FPTYPE *rab, FPTYPE *drhocg, 
+        const int mesh, const int igl0, const int ngg, const double omega
+){
+    const double FOUR_PI =  4.0 * 3.14159265358979323846;
+
+    int idx = threadIdx.x + blockIdx.x * blockDim.x;
+
+    FPTYPE aux_d[2];
+    FPTYPE rhocg1=0.0, f_0=0.0, f_2=0.0, f_1=0.0;
+
+    if (idx >= ngg) {return;}
+    
+    for( int ir = 0;ir< mesh; ir++)
+    {
+        const int ir_2 = ir%2;
+        const FPTYPE gx_r = gx_arr[idx] * r [ir];
+        aux_d [ir_2]  = r [ir] * rhoc [ir] * (r [ir] * cos (gx_r) / gx_arr[idx] - sin (gx_r) / gx_arr[idx] * gx_arr[idx]);
```

Comment:
"sin (gx_r) / gx_arr[idx] * gx_arr[idx]" should be "sin (gx_r) / (gx_arr[idx] * gx_arr[idx])".